From 1abc413af44652d6a76d5b5c2afe90788595008e Mon Sep 17 00:00:00 2001
From: Jonathan Bell <jonathan@raspberrypi.com>
Date: Mon, 15 Jul 2024 13:57:01 +0100
Subject: [PATCH 1169/1215] mmc: brcmstb: don't squash card-busy detection on
 bcm2712

Commit 485d9421719b  ("mmc: sdhci-brcmstb: check R1_STATUS for
erase/trim/discard") introduced a new flag and defaulted to disabling
card busy detection across all platforms with this controller.

This is required for IO voltage switching, as the card drives CMD low
while the switch is in progress.

Signed-off-by: Jonathan Bell <jonathan@raspberrypi.com>
---
 drivers/mmc/host/sdhci-brcmstb.c | 1 +
 1 file changed, 1 insertion(+)

--- a/drivers/mmc/host/sdhci-brcmstb.c
+++ b/drivers/mmc/host/sdhci-brcmstb.c
@@ -430,6 +430,7 @@ static const struct brcmstb_match_priv m
 };
 
 static const struct brcmstb_match_priv match_priv_2712 = {
+	.flags = BRCMSTB_MATCH_FLAGS_USE_CARD_BUSY,
 	.hs400es = sdhci_brcmstb_hs400es,
 	.cfginit = sdhci_brcmstb_cfginit_2712,
 	.ops = &sdhci_brcmstb_ops_2712,
